src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -579,15 +689,101 @@ protected void deleteLoadBalancer(String elbName) {
         JcloudsLocation loc = getLocation();
         LOG.debug("Deleting ELB: "+elbName);
 
-        AmazonElasticLoadBalancingClient client = newClient(loc);
+        AmazonElasticLoadBalancingClient client = elbClient(loc);
         try {
             DeleteLoadBalancerRequest deleteLoadBalancerRequest = new DeleteLoadBalancerRequest(elbName);
             client.deleteLoadBalancer(deleteLoadBalancerRequest);
+            removeSecurityGroups();
         } finally {
             if (client != null) client.shutdown();
         }
     }
 
+    private void removeSecurityGroups() {
+        Collection<String> securityGroups = sensors().get(LOAD_BALANCER_SECURITY_GROUPS);
+
+        final AmazonEC2Client client = ec2Client(getLocation());
+        if (securityGroups != null) {
+            List<Task> removals = MutableList.of();
+            for (String groupId : securityGroups) {
+                removals.add(removeOneSecurityGroup(client, groupId));
+            }
+            ParallelTask<Boolean> invoke = new ParallelTask<>(MutableMap.of(
```

Comment:
I have a personal preference for creating tasks by using `Tasks.builder()`. It codifies the arguments that you've given in the map.